### PR TITLE
Removed code that sets default values in threadsafe_state_storage.cpp to robot state and added unittests

### PR DESCRIPTION
**Pull request by pradeepr-roboticist**
### Description

This issue is documented in #453 .

I removed line 66 of  /moveit_planners/ompl/ompl_interface/src/detail/threadsafe_state_storage.cpp

`st->setToDefaultValues();`

It seems to me that this line should not be there as it negates intention of intializing the robot state the line before:
`
st = new moveit::core::RobotState(start_state_)
`

I have also added a unit-test for this.

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [x] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by henningkayser — COMMENTED** [review]
Thanks a lot for debugging this issue! I think you might be spot on with how the joint values of the other groups are being reset. I'd like to know why this was added in the first place and will test this before merging.

**Review by henningkayser — APPROVED** [review]
Confirmed fixed!

**Review by JafarAbdi — APPROVED** [review]
This also fixes the bug in the move_group demo where planning for the hand group always fails, thanks!

Could you please run the [pre-commit checks](https://moveit.ros.org/documentation/contributing/code/#pre-commit-formatting-checks) .?

**Comment by pradeepr-roboticist** [comment]
> This also fixes the bug in the move_group demo where planning for the hand group always fails, thanks!
> 
> Could you please run the [pre-commit checks](https://moveit.ros.org/documentation/contributing/code/#pre-commit-formatting-checks) .?

Thanks! I just ran the files through the pre-commit checks.

File: moveit_planners/ompl/ompl_interface/test/test_threadsafe_state_storage.cpp
Comment (prajd3d) [reply moveit_planners/ompl/ompl_interface/test/test_threadsafe_state_storage.cpp:4]: Just fixed. :)

File: moveit_planners/ompl/ompl_interface/test/test_threadsafe_state_storage.cpp
Comment (prajd3d) [reply moveit_planners/ompl/ompl_interface/test/test_threadsafe_state_storage.cpp]: Just changed.

**Comment by prajd3d** [comment]
Please let me know if there are next steps to get this merged. Thanks!

**Comment by prajd3d** [comment]
@JafarAbdi  This is my first PR in MoveIt2 and I was wondering if you needed anything from me regarding this PR. Please let me know. Thanks!

**Comment by henningkayser** [comment]
> @JafarAbdi This is my first PR in MoveIt2 and I was wondering if you needed anything from me regarding this PR. Please let me know. Thanks!

It's all good @prajd3d, sorry for letting you wait. I was hoping to get more clarification for the affected line in discussions outside of this PR, but if there are really issues with uninitialized memory in the start state, I'm sure there would be better ways to fix that. I'll merge this PR once CI is happy.

**Comment by prajd3d** [comment]
> > @JafarAbdi This is my first PR in MoveIt2 and I was wondering if you needed anything from me regarding this PR. Please let me know. Thanks!
> 
> It's all good @prajd3d, sorry for letting you wait. I was hoping to get more clarification for the affected line in discussions outside of this PR, but if there are really issues with uninitialized memory in the start state, I'm sure there would be better ways to fix that. I'll merge this PR once CI is happy.

Awesome! Thanks for your time!

File: moveit_planners/ompl/ompl_interface/src/detail/threadsafe_state_storage.cpp
Comment (JeroenDM) [reply moveit_planners/ompl/ompl_interface/src/detail/threadsafe_state_storage.cpp:66]: @henningkayser sorry for being late to the party :( I added this line last summer while debugging nasty memory issues. (probably this one #2239). Valgrind reported uninitialized memory here so I changed it. Looking at the `robot_state.cpp` code now I cannot see how this could have caused an issue.

My best guess is that I never removed it (or really understood it) because of my limited experience at the time and it was not catched in a review because CI did not complain.

Comment (henningkayser) [reply moveit_planners/ompl/ompl_interface/src/detail/threadsafe_state_storage.cpp:66]: @JeroenDM thanks for responding anyway! ;) While removing this line fixes the linked issues, I think that we are still dealing with some memory issues. So maybe you were onto something there. At least we've seen flaky constrained planner tests and some uninitialized memory during the tests as well. We also haven't looked into this too much yet. Good that this is clarified, though.

**Pull request merged**
